### PR TITLE
Add feature to allow License to be stored in AWS Systems Manager Parameter Store

### PR DESCRIPTION
**Pull request by ovlords**

### CODE REVIEW

File: conf/ansible/inventory/group_vars/defaults.yaml [hunk 1]
```diff
@@ -64,6 +64,8 @@ aem:
     version: 4.3.1
   artifacts_base: overwrite-me
   enable_custom_image_provisioner: false
+  aem_license_sm_parameter: overwrite-me
```
Comment (cliffano) [inline conf/ansible/inventory/group_vars/defaults.yaml]: Also, why is this parameter under `aem` section when the relevancy is specific to `aws`?
Shouldn't this be under `aws` ?

File: provisioners/puppet/modules/config/manifests/license.pp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  $region,
+) {
+
+  # TODO: This code is a workaround, this needs a better conversion that can map any aem_profile into any aem_version
+  case $aem_profile {
+    /^aem62/:  { $aem_version = '6.2' }
+    /^aem63/:  { $aem_version = '6.3' }
+    /^aem64/:  { $aem_version = '6.4' }
+    /^aem65/:  { $aem_version = '6.5' }
+    default: { $aem_version = 'default' }
```
[the remark this answers is not in the document]
Comment (ovlords) [reply provisioners/puppet/modules/config/manifests/license.pp]: Altered to 6.4

File: provisioners/puppet/modules/config/manifests/license.pp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  exec { "Create ${tmp_dir}/license":
```
[the remark this answers is not in the document]
Comment (ovlords) [reply provisioners/puppet/modules/config/manifests/license.pp]: All resources are kept in their own folders.  It is consistent with other similar resources such as the 'certs' resources.

[the remark this answers is not in the document]
Comment (ovlords) [reply provisioners/puppet/modules/config/manifests/license.pp]: Altered as requested.

File: conf/ansible/inventory/group_vars/defaults.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ovlords) [reply conf/ansible/inventory/group_vars/defaults.yaml]: I will rename it to aem_license.

Comment (ovlords) [reply conf/ansible/inventory/group_vars/defaults.yaml]: It will be moved to the aws section.

**Pull request merged**